src/common/libsubprocess/remote.c

```diff
@@ -640,6 +476,58 @@ static int remote_output_buffered (flux_subprocess_t *p,
     return 0;
 }
 
+/* In the event channel had data / closed before process running */
+static int send_channel_data (flux_subprocess_t *p)
+{
+    struct subprocess_channel *c;
+    c = zhash_first (p->channels);
+    while (c) {
+        if (fbuf_bytes (c->write_buffer) > 0) {
+            const char *ptr;
+            int len;
+            if (!(ptr = fbuf_read (c->write_buffer, -1, &len))) {
+                llog_debug (p,
+                            "error reading buffered data: %s",
+                            strerror (errno));
+                set_failed (p, "internal fbuf_read error");
+                return -1;
+            }
+            if (subprocess_write (p->h,
+                                  p->service_name,
+                                  p->rank,
+                                  p->pid,
+                                  c->name,
+                                  ptr,
+                                  len,
+                                  false) < 0) {
+                llog_debug (p,
+                            "error sending rexec.write request: %s",
+                            strerror (errno));
+                set_failed (p, "internal close error");
+                return -1;
+            }
+        }
+        if (c->closed) {
+            if (subprocess_write (p->h,
+                                  p->service_name,
+                                  p->rank,
+                                  p->pid,
+                                  c->name,
+                                  NULL,
+                                  0,
+                                  true) < 0) {
```

Comment:
Could this be combined with the `subprocess_write()` call above?  E.g.
```c
if (c->closed || fbuf_bytes (c->write_buffer) > 0) {
    ...
}
```

Reply:
ahhh i think you're right.  It would have been two calls if the user did it, but we can combine it at this point in the code.